src/policy/semantic.rs

```diff
@@ -569,15 +592,18 @@ impl<Pk: MiniscriptKey> Policy<Pk> {
 
     /// Filter a policy by eliminating absolute timelock constraints
     /// that are not satisfied at the given `n` (`n OP_CHECKLOCKTIMEVERIFY`).
-    pub fn at_lock_time(mut self, n: u32) -> Policy<Pk> {
+    pub fn at_lock_time(mut self, n: LockTime) -> Policy<Pk> {
         self = match self {
             Policy::After(t) => {
-                if !timelock::absolute_timelocks_are_same_unit(t, n) {
+                // FIXME: If the point of the LockTime type is to make it hard to make mistakes then
+                // this code shows it fails at that since we _must_ check is_same_unit before comparing.
```

Comment:
Indeed, sorry I was absent during the review of the initial code. I get it that there was some hesitance in implementing `PartialOrd`, but why not have a method that returns Option<cmp::Ordering>? 

We should really not need to check that they are same units here.

Reply:
`PartialOrd` gives us a method that returns `Option<cmp::Ordering>`. It's `Ord` that we couldn't implement, because there is no natural total ordering on timelocks, but which we wanted to implement so that you could use `Vec::sort` and derive `Ord` on structures that contain timelocks.

Reply:
I think @sanket1729 means that `PartialOrd` still allows `<` and `>` operators which are foot guns. Maybe this should be a `clippy` lint.